src/radius_attr.rs

```diff
@@ -83,88 +91,133 @@ pub enum FramedCompression {
+            Ipv4Addr::new(v[0], v[1], v[2], v[3])
+        )),
         5 => map!(i, be_u32, |v| RadiusAttribute::NasPort(v)),
-        6 => map!(i, be_u32, |v| RadiusAttribute::ServiceType(v)),
-        7 => map!(i, be_u32, |v| RadiusAttribute::FramedProtocol(v)),
-        8 => map!(i, take!(4), |v:&[u8]| RadiusAttribute::FramedIPAddress(Ipv4Addr::new(v[0],v[1],v[2],v[3]))),
-        9 => map!(i, take!(4), |v:&[u8]| RadiusAttribute::FramedIPNetmask(Ipv4Addr::new(v[0],v[1],v[2],v[3]))),
-        10 => map!(i, be_u32, |v| RadiusAttribute::FramedRouting(v)),
+        6 => map_opt!(i, be_u32, |v| ServiceType::from_u32(v)
+            .map(|v| RadiusAttribute::ServiceType(v))),
+        7 => map_opt!(i, be_u32, |v| FramedProtocol::from_u32(v)
+            .map(|v| RadiusAttribute::FramedProtocol(v))),
+        8 => map!(i, take!(4), |v: &[u8]| RadiusAttribute::FramedIPAddress(
+            Ipv4Addr::new(v[0], v[1], v[2], v[3])
+        )),
+        9 => map!(i, take!(4), |v: &[u8]| RadiusAttribute::FramedIPNetmask(
+            Ipv4Addr::new(v[0], v[1], v[2], v[3])
+        )),
+        10 => map_opt!(i, be_u32, |v| FramedRouting::from_u32(v)
+            .map(|v| RadiusAttribute::FramedRouting(v))),
         11 => value!(i, RadiusAttribute::FilterId(i)),
         12 => map!(i, be_u32, |v| RadiusAttribute::FramedMTU(v)),
-        13 => map!(i, be_u32, |v| RadiusAttribute::FramedCompression(v)),
-        _ => value!(i, RadiusAttribute::Unknown(t,i)),
+        13 => map_opt!(i, be_u32, |v| FramedCompression::from_u32(v)
+            .map(|v| RadiusAttribute::FramedCompression(v))),
+        26 => {
+            if i.len() < 5 {
+                return IResult::Incomplete(Needed::Size(5));
+            }
+            value!(
+                i,
+                RadiusAttribute::VendorSpecific(
+                    ((i[0] as u32) << 24) + ((i[1] as u32) << 16) + ((i[2] as u32) << 8)
+                        + (i[3] as u32),
+                    &i[4..]
+                )
+            )
+        }
+        _ => value!(i, RadiusAttribute::Unknown(t, i)),
     }
 }
 
-pub fn parse_radius_attribute(i:&[u8]) -> IResult<&[u8],RadiusAttribute> {
-    do_parse!(i,
-        t: be_u8 >>
-        l: verify!(be_u8, |val:u8| val >= 2) >>
-        v: flat_map!(take!(l-2),call!(parse_attribute_content,t)) >>
-        ( v )
+pub fn parse_radius_attribute(i: &[u8]) -> IResult<&[u8], RadiusAttribute> {
+    do_parse!(
+        i,
+        t: be_u8 >> l: verify!(be_u8, |val: u8| val >= 2)
+            >> v: flat_map!(take!(l - 2), call!(parse_attribute_content, t)) >> (v)
     )
 }
 
 #[cfg(test)]
 mod tests {
     use radius_attr::*;
-    use nom::{IResult,ErrorKind};
-
-#[test]
-fn test_attribute_invalid() {
-    let data = &[255, 0, 2, 2];
-    assert_eq!(
-        parse_radius_attribute(data),
-        IResult::Error(error_position!(ErrorKind::Verify,&data[1..]))
-    );
-}
+    use nom::{ErrorKind, IResult};
+
+    #[test]
```

Comment:
This section seems indented by tabs - or am I mistaken by the color ? This seems also wrong imho